packages/astro/src/vite-plugin-mdx/index.ts

```diff
@@ -1,99 +1,19 @@
-import type { TransformResult } from 'rollup';
-import { type Plugin, type ResolvedConfig, transformWithEsbuild } from 'vite';
-import type { AstroRenderer, AstroSettings } from '../@types/astro.js';
-import type { Logger } from '../core/logger/core.js';
-import type { PluginMetadata } from '../vite-plugin-astro/types.js';
-
-import babel from '@babel/core';
+import { type Plugin, transformWithEsbuild } from 'vite';
 import { CONTENT_FLAG, PROPAGATED_ASSET_FLAG } from '../content/index.js';
 import { astroEntryPrefix } from '../core/build/plugins/plugin-component-entry.js';
 import { removeQueryString } from '../core/path.js';
-import tagExportsPlugin from './tag.js';
-
-interface TransformJSXOptions {
-	code: string;
-	id: string;
-	mode: string;
-	renderer: AstroRenderer;
-	ssr: boolean;
-	root: URL;
-}
-
-async function transformJSX({
```

Comment:
This function is moved to `./transform-jsx.ts` and is updated to retrieve the transform options differently.